.github/workflows/main.yaml

```diff
@@ -0,0 +1,89 @@
+name: Quality and Build 
+
+on:
+  pull_request:
+    branches: [main]
+    types: [opened, synchronize, reopened, labeled]
+  push:
+    branches: [main]
+
+jobs:
+  lint:
+    runs-on: ubuntu-20.04
+    name: Linting
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.7.1'
+
+      - name: Setup NodeJS
+        uses: actions/setup-node@v4
+        with:
+          node-version: '18'
+
+      - name: Install dependencies
+        run: |
+          node -v && yarn -v && yarn install --immutable && \
+          git status
+
+      - name: Prettier
+        run: |
+          yarn format:check
+
+      - name: Linting
+        run: |
+          yarn lint:check
+
+  test:
+    runs-on: ubuntu-20.04
+    name: Testing
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v4
```

Comment:
needed?

Reply:
Yes, a specific version of Python is needed for `ffi-napi` and `ref-napi` build process, also wanted to match the Dockerfile version